dpctl-capi/source/dpctl_sycl_platform_interface.cpp

```diff
@@ -84,6 +84,9 @@ __dpctl_give DPCTLSyclPlatformRef DPCTLPlatform_CreateFromSelector(
             PRef = wrap(P);
         } catch (std::bad_alloc const &ba) {
             std::cerr << ba.what() << '\n';
+        } catch (runtime_error const &re) {
+            std::cerr << re.what() << '\n';
+            return nullptr;
```

Comment:
Please, do `catch(...)` finally. Any uncatched exception will break C API interface.

Reply:
Let us do that as a follow up along with the overall work needed to improve error handling (#35).